src/CodeGen_C.cpp

```diff
@@ -2458,6 +2476,110 @@ void CodeGen_C::visit(const Call *op) {
+        internal_assert(op->args.size() >= 2);
+
+        string type_carrier = print_expr(op->args[0]);
+        const int64_t *count_ptr = as_const_int(op->args[1]);
+        internal_assert(count_ptr != nullptr);
+        int64_t count = *count_ptr;
+
+        // Get the args
+        vector<string> values;
+        for (size_t i = 2; i < op->args.size(); i++) {
+            values.push_back(print_expr(op->args[i]));
+        }
+        string struct_name = unique_name('s');
+        string array_specifier;
+        if (count > 0) {
+            array_specifier = "[]";
+        }
+        stream << get_indent() << "std::remove_pointer<decltype(" << type_carrier << ")>::type " << struct_name << array_specifier << " = {\n";
+        // List the values.
+        indent++;
+        int item = 0;
+        size_t initializer_count = values.size() / count;
+        internal_assert((values.size() % count) == 0);
+        do {
+            if (count > 0) {
+                stream << get_indent() << "{\n";
+                indent++;
+            }
+            for (size_t i = 0; i < initializer_count; i++) {
+                stream << get_indent() << values[i + item * initializer_count];
+                if (i < op->args.size() - 1) {
+                    stream << ",";
+                }
+                stream << "\n";
+            }
+            if (count > 0) {
+                indent--;
+                stream << get_indent() << "},\n";
+            }
+            item += 1;
+        } while (item < count);
+        indent--;
+        stream << get_indent() << "};\n";
+        // Return a pointer to it of the appropriate type
+
+        // TODO: This is dubious type-punning. We really need to
+        // find a better way to do this. We dodge the problem for
+        // the specific case of buffer shapes in the case above.
+        if (op->type.handle_type) {
+            rhs << "(" << print_type(op->type) << ")";
+        }
+        rhs << "(&" << struct_name << ((count > 0) ? "[0])" : ")");
+    } else if (op->is_intrinsic(Call::load_struct_member)) {
+        internal_assert(op->args.size() == 3);
+        const uint64_t *index = as_const_uint(op->args[2]);
+        internal_assert(index != nullptr);
+        std::string struct_base = print_expr(op->args[0]);
+        std::string type_carrier = print_expr(op->args[1]);
+        rhs << "((decltype(" << type_carrier << "))" << struct_base << ")->"
+            << "f_" << *index;
+    } else if (op->is_intrinsic(Call::resolve_function_name)) {
+        internal_assert(op->args.size() == 1);
+        const Call *decl_call = op->args[0].as<Call>();
+        internal_assert(decl_call != nullptr);
+        std::string c_name = c_print_name(decl_call->name, false);
+        rhs << "(&" << c_name << ")";
+    } else if (op->is_intrinsic(Call::get_user_context)) {
+        internal_assert(op->args.empty());
+        rhs << "_ucon";
+    } else if (op->is_intrinsic(Call::get_pointer_symbol_or_null)) {
```

Comment:
The LLVM implementation of this symbol attempts a symbol lookup, but this just returns null unconditionally, which is mighty puzzling. Is it a bug, or do I just misunderstand the usage? (Documentation somewhere would be helpful)